c/authService.c

```diff
@@ -173,16 +188,209 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
     respondWithError(res, HTTP_STATUS_BAD_REQUEST, "Unexpected access level");
     return 0;
   }
-  /* printf("query: user %s, class %s, entity %s, access %d\n", userName, class,
-      entity, access); */
+  // printf("\n\nquery: user %s, class %s, entity %s, access %d accessStr %s\n", userName, class,
+      // entity, access, accessStr);
   privilegedServerName = getConfiguredProperty(service->server,
       HTTP_SERVER_PRIVILEGED_SERVER_PROPERTY);
   rc = zisCheckEntity(privilegedServerName, userName, class, entity, access,
       &reqStatus);
+      // printf("\n\nprivileged server name: %s", privilegedServerName);
   respond(res, rc, &reqStatus);
   return 0;
 }
 
+const char* getProfileNameFromRequest(char *url, char *method, int instanceID) {
+  char type[8]; // core || config || service
+  char productCode[1024];
+  char rootServiceName[1024];
+  char subUrl[15][1024];
+  char *profileName = (char*) safeMalloc(1024, "profileName");
```

Comment:
* No need to case
* A null check is needed

Reply:
Wdym by null check? I don't see this in the other usages of safeMalloc, sorry new to C